core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -1291,15 +1305,12 @@ protected ClientTransportFactory buildTransportFactory() {
       builder.proxyDetector = nameResolverArgs.getProxyDetector();
       builder.defaultPort = nameResolverArgs.getDefaultPort();
       builder.userAgent = userAgent;
-      return
-          new ManagedChannelImpl(
-              builder,
-              transportFactory,
-              backoffPolicyProvider,
-              balancerRpcExecutorPool,
-              stopwatchSupplier,
-              Collections.<ClientInterceptor>emptyList(),
-              timeProvider);
+      return builder;
+    }
+
+    @Override
+    public ManagedChannel createResolvingOobChannel(String target) {
```

Comment:
Do you still need this overriding? Same for below.

Reply:
For below we definitely need.

Reply:
done. for the Forwarding one, i didn't change because the method is not final.